### ISSUE TITLE
Each connection establishment will be done at localhost

### ISSUE THREAD

**Issue by mlang-de** [digitalpetri/stack-bench#2]
Regardlessly which address is entered into the config file (e.g. "opc.tcp://10.11.5.8:51210/UA/Server"), the client will connect to "opc.tcp://localhost:51210/UA/Server". This is caused by function "getOpcUaClient(endpointUrl: String)". Variable 'endpointUrl' contains the correct string from the config file, after UaTcpStackClient.getEndpoints(endpointUrl).get() the IP address in variable 'endpoints' will be replaced and the client tries to connect to the localhost.

If the server certificate is modified with the current fix IP address the stack-bench tool will work.

**Comment by kevinherron** [comment]
Is the server returning endpoints that contain localhost in the endpoint URL rather than the IP address?

**Comment by kevinherron** [comment]
If so, this is a common question I've covered here: http://stackoverflow.com/questions/40554198/java-opc-ua-client-eclipse-milo-endpoint-url-changes-to-localhost/40555243#40555243

In this case though, you don't actually want to fix anything. The stack-bench tool should only be run on localhost.

**Comment by mlang-de** [comment]
You are right! I use the OPC Foundation Sample Server and check the GetEndpointsResponse with wireshark. The received EndpointUrl is 'opc.tcp://localhost:51210/UA/SampleServer'. 

One other (dirty?) way to connect with the remote server could be to modify the config.xml from the OPCF sample server. 
/* original */
   <d3p1:String>opc.tcp://localhost:61210/UA/SampleClient</d3p1:String>
/* modification for IP: 192.168.254.160 */
   <d3p1:String>opc.tcp://192.168.254.160:61210/UA/SampleClient</d3p1:String>

------------------------------------------
"The stack-bench tool should only be run on localhost."
Why this? Are there some technical reasons?

**Comment by kevinherron** [comment]
If you don't run on localhost I suspect most servers will show the same performance numbers as you're limited by the latency of the network connection.
